Content.Server/Xenoarchaeology/XenoArtifacts/Triggers/Systems/ArtifactInteractionTriggerSystem.cs

```diff
@@ -1,27 +1,45 @@
 using Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Components;
-using Content.Shared.ActionBlocker;
 using Content.Shared.Interaction;
 using Content.Shared.Interaction.Helpers;
-using Robust.Shared.GameObjects;
-using Robust.Shared.IoC;
+using Content.Shared.Physics.Pull;
+using Content.Shared.Weapons.Melee;
 
 namespace Content.Server.Xenoarchaeology.XenoArtifacts.Triggers.Systems;
 
-public class ArtifactInteractionTriggerSystem : EntitySystem
+public sealed class ArtifactInteractionTriggerSystem : EntitySystem
 {
     [Dependency] private readonly ArtifactSystem _artifactSystem = default!;
 
     public override void Initialize()
     {
         base.Initialize();
+        SubscribeLocalEvent<ArtifactInteractionTriggerComponent, PullStartedMessage>(OnPull);
+        SubscribeLocalEvent<ArtifactInteractionTriggerComponent, AttackedEvent>(OnAttack);
         SubscribeLocalEvent<ArtifactInteractionTriggerComponent, InteractHandEvent>(OnInteract);
     }
 
+    private void OnPull(EntityUid uid, ArtifactInteractionTriggerComponent component, PullStartedMessage args)
+    {
+        if (!component.PullActivation)
+            return;
+
+        _artifactSystem.TryActivateArtifact(uid, args.Puller.Owner);
+    }
+
+    private void OnAttack(EntityUid uid, ArtifactInteractionTriggerComponent component, AttackedEvent args)
+    {
+        if (!component.AttackActivation)
+            return;
+
+        _artifactSystem.TryActivateArtifact(uid, args.User);
+    }
+
     private void OnInteract(EntityUid uid, ArtifactInteractionTriggerComponent component, InteractHandEvent args)
     {
-        if (args.Handled)
+        if (args.Handled || !args.InRangeUnobstructed())
             return;
```

Comment:
does interactionsystem not check `InRangeUnobstructed` itself?

Reply:
Now yes, it was deleted by recent interaction clean-up